src/components/mapComponents/maps/treeMap/index.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import {
+  InitMapData,
+  MapViews,
+  NeighborhoodGeoData,
+  ReturnMapData,
+  SiteGeoData,
+} from '../../ducks/types';
+import { NO_SITE_SELECTED } from '../../../treePopup';
+import { BOSTON } from '../../constants';
+import { addHandleZoomChange, getImageSize } from '../../logic/event';
+import { initSiteView } from '../../logic/init';
+import { CheckboxValueType } from 'antd/es/checkbox/Group';
+import { setSitesStyle } from '../../logic/style';
+import SiteLegend from '../../mapPageComponents/siteLegend';
+import { MapStateProps, Routes } from '../../../../App';
+import MapWithPopup from '../mapWithPopup';
+
+interface TreeMapProps {
+  readonly neighborhoods: NeighborhoodGeoData;
+  readonly sites: SiteGeoData;
+  readonly mobile: boolean;
+  readonly returnTo?: Routes;
+}
+
+const TreeMap: React.FC<TreeMapProps> = ({
+  neighborhoods,
+  sites,
+  mobile,
+  returnTo,
+}) => {
+  const location = useLocation<MapStateProps>();
+
+  const [loadedMapData, setLoadedMapData] = useState<ReturnMapData>();
+
+  let defaultZoom = 12;
+  let defaultCenter = BOSTON;
+  if (location.state) {
+    defaultZoom = location.state.zoom;
+    defaultCenter = { lat: location.state.lat, lng: location.state.lng };
+  }
+
+  const initMap = (mapData: InitMapData): ReturnMapData => {
+    // A marker to show at the location a user searches for
+    const searchMarker = new google.maps.Marker({
+      map: mapData.map,
+    });
+    const mapLayersAndListeners = initSiteView(mapData, neighborhoods, sites);
+
+    const setMapData: ReturnMapData = {
+      map: mapData.map,
+      zoom: mapData.zoom,
+      privateStreetsLayer: mapLayersAndListeners.privateStreetsLayer,
+      neighborhoodsLayer: mapLayersAndListeners.neighborhoodsLayer,
+      sitesLayer: mapLayersAndListeners.sitesLayer,
+      searchMarker,
+      zoomListener: mapLayersAndListeners.zoomListener,
+      markersArray: mapData.markersArray,
+    };
+
+    setLoadedMapData(setMapData);
+
+    return setMapData;
+  };
+
+  // Add new zoom listener and update sites style whenever visibleSites changes
+  const onCheck = (values: CheckboxValueType[]): void => {
```

Comment:
❓  Will this zoom functionality be dramatically different for blocks? it seems like the two map pages will have almost identical code with just a couple of different parameters/props.

Reply:
Yeah, most of the zoom logic is abstracted to `addHandleZoomChange` (which this function uses). This function is unique because when we hide some sites, we have to remove and re-add the zoom handler, then turn on the site styling if we're already zoomed in (which the handler doesn't initially check for).